README.md

```diff
@@ -33,75 +33,112 @@ Confused about the terminology of APM? Take a look at the [APM Glossary][visuali
 
-    $ hooks/autohook.sh install
+For local development, the repository includes these same checks as custom
+scripts that run as Git pre-commit hooks.
+
+To enable all pre-commit hooks in your cloned repository, run the following command:
 
-### Set up your environment
+    $ hooks/autohook.sh install
+    
+[pre_check]: https://github.com/DataDog/dd-trace-py/blob/5b97489e2b073fa773819904cdef26981a5a28df/.circleci/config.yml#L293-L315
 
-#### Set up docker
+### Development environment
 
-The test suite requires many backing services such as PostgreSQL, MySQL, Redis
-and more. We use `docker` and `docker-compose` to run the services in our CI
-and for development. To run the test matrix, please [install docker][docker] and
-[docker-compose][docker-compose] using the instructions provided by your platform. Then
-launch them through:
+#### Docker
 
-    $ docker-compose up -d
+Install [Docker](docker) on your development environment for running tests
+locally.
 
 [docker]: https://www.docker.com/products/docker
-[docker-compose]: https://www.docker.com/products/docker-compose
 
-#### Set up Python
+#### Python
 
-1. Clone the repository locally: `git clone https://github.com/DataDog/dd-trace-py`
-2. The tests for this project run on various versions of Python. We recommend
-   using a Python version management tool, such as
-   [pyenv](https://github.com/pyenv/pyenv), to utilize multiple versions of
-   Python. Install Pyenv: https://github.com/pyenv/pyenv#installation
-3. Install the relevant versions of Python in Pyenv: `pyenv install 3.9.1, 2.7.18, 3.5.10, 3.6.12, 3.7.9, 3.8.7, 3.10.0`
-4. Make those versions available globally: `pyenv global 3.9.1, 2.7.18, 3.5.10, 3.6.12, 3.7.9, 3.8.7, 3.10.0`
+The library supports multiple Python runtime versions and platforms. See [supported runtimes](https://ddtrace.readthedocs.io/en/stable/versioning.html#supported-runtimes) for the full list. You have two options for preparing a local development environment. You can use pyenv to install Python runtimes on your local machine or you can use the `datadog/dd-trace-py` Docker image which includes all runtimes.
 
-### Testing
+##### Option #1: pyenv
+
+1. [Install pyenv](https://github.com/pyenv/pyenv#getting-pyenv)
+2. Install Python runtimes with pyenv
+3. [Optional] [Install pyenv-virtualenv](https://github.com/pyenv/pyenv-virtualenv)
+
+To install all the tested Python runtimes:
 
-#### Running Tests in docker
+    cd docker; pyenv local | xargs -L 1 pyenv install
+    
+We recommend using a virtual environment for local development. 
 
-Once your docker-compose environment is running, you can use the shell script to
-execute tests within a Docker image. You can start the container with a bash shell:
+If you followed the optional step to install the `pyenv-virtualenv` plugin, you can use it to manage your Python virtual environments:
 
-    $ scripts/ddtest
+    pyenv virtualenv 3.9.11 ddtrace-py3.9
+    pyenv activate ddtrace-py3.9
```

Comment:
Should we emphasize that py3.9 is the standard to run the tests? if you run `riot` and `mypy` with `riot run -s mypy` it fails in py3.8. I get this error below with autohook:

```
docs/conf.py:286: error: "dict" is not subscriptable, use "typing.Dict" instead  [misc]
docs/conf.py:422: error: "list" is not subscriptable, use "typing.List" instead  [misc]
docs/conf.py:422: error: "tuple" is not subscriptable, use "typing.Tuple" instead  [misc]
docs/conf.py:434: error: "list" is not subscriptable, use "typing.List" instead  [misc]
docs/conf.py:434: error: "tuple" is not subscriptable, use "typing.Tuple" instead  [misc]
docs/conf.py:473: error: "list" is not subscriptable, use "typing.List" instead  [misc]
docs/conf.py:473: error: "tuple" is not subscriptable, use "typing.Tuple" instead  [misc]
docs/conf.py:509: error: "list" is not subscriptable, use "typing.List" instead  [misc]
docs/conf.py:530: error: "list" is not subscriptable, use "typing.List" instead  [misc]
docs/conf.py:531: error: "list" is not subscriptable, use "typing.List" instead  [misc]
docs/conf.py:595: error: "dict" is not subscriptable, use "typing.Dict" instead  [misc]
ddtrace/settings/config.py:22: error: "set" is not subscriptable, use "typing.Set" instead  [misc]
ddtrace/profiling/_asyncio.py:29: error: Unused "type: ignore" comment
ddtrace/profiling/_asyncio.py:38: error: Unused "type: ignore" comment
Found 14 errors in 3 files (checked 389 source files)
```